### PR TITLE
enable autobuild on PRs and add "publish" github workflow

### PR DESCRIPTION
**Pull request by screret**

### CODE REVIEW

File: .github/workflows/auto-build.yml [hunk 1]
```diff
@@ -4,8 +4,10 @@ on:
   push:
     branches:
       - 'master'
-    paths:
-      - 'gradle.properties'
+  pull_request:
+    types:
+      - labeled
+      - synchronize
```
Comment (Yefancy) [inline .github/workflows/auto-build.yml:11]: remove path is fine. but building for all `pull_request` will cause too much actions I think?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          CHANGELOG_LOCATION: "${{env.CHANGELOG_LOCATION}}"
+          RELEASE_TYPE: "${{env.RELEASE_TYPE}}"
+        with:
+          arguments: 'modrinth --daemon'
+          generate-job-summary: false
+          gradle-home-cache-includes: |
+            caches
+            jdks
+            notifications
+            wrapper
```
Comment (Yefancy) [inline .github/workflows/publish.yml:89]: Now let's comment them out until when we relesase. btw, we can also use the Kir plugin which will automatically handle cf, modrinth and github releases. It's very convenient

**Review by Yefancy — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-build.yml (hunk 1 above)
Comment (screret) [reply .github/workflows/auto-build.yml:11]: I tried to make it only work for tagged PRs
Guess it didn't :shrug:

File: .github/workflows/publish.yml (hunk 2 above)
Comment (screret) [reply .github/workflows/publish.yml:89]: Sounds good, i just copied that off 1.12

File: .github/workflows/auto-build.yml (hunk 1 above)
Comment (screret) [reply .github/workflows/auto-build.yml:11]: Sorry, i meant labeled as `Deployment: artifact`

**Comment by Yefancy** [comment]
I will mannually set it  later

**Pull request closed without merging**